src/components/entity/ha-entity-picker.js

```diff
@@ -136,8 +136,8 @@ class HaEntityPicker extends EventsMixin(LocalizeMixin(PolymerElement)) {
     let entityIds = Object.keys(hass.states);
 
     if (domainFilter) {
-      entityIds = entityIds.filter(
-        (eid) => eid.substr(0, eid.indexOf(".")) === domainFilter
+      entityIds = entityIds.filter((eid) =>
+        domainFilter.includes(eid.substr(0, eid.indexOf(".")))
```

Comment:
We should reverse this change, as `includes` will also match if it's not an exact match. So if we ever get a media component, it would match all media player components.